app/forms/otp_verification_form.rb

```diff
@@ -1,31 +1,65 @@
 class OtpVerificationForm
+  include ActiveModel::Model
+
+  validates :code, presence: true
+  validate :validate_code_length
+  validate :validate_code_matches_format
+  validate :validate_user_otp_presence
+  validate :validate_user_otp_expiration
+  validate :validate_code_equals_user_otp
+
   def initialize(user, code)
     @user = user
     @code = code
   end
 
   def submit
+    success = valid?
+
+    user.clear_direct_otp if success
```

Comment:
Wondering if it makes sense for this to happen here in the form, or if it would be better in the controller?